src/operator/OrgOverlay.tsx

```diff
@@ -277,6 +277,41 @@ const OrgOverlay: FC = () => {
                       />
                     </Grid.Column>
                   </Grid.Row>
+                  {limits?.timeout && (
+                    <Grid.Row>
+                      <Grid.Column widthMD={Columns.Four}>
+                        <Form.Label label="Unconditional Timeout (seconds)" />
+                        <LimitsField
+                          type={InputType.Number}
+                          name="timeout.queryUnconditionalTimeoutSeconds"
+                          limits={limits}
+                          onChangeLimits={setLimits}
+                        />
+                      </Grid.Column>
+                      <Grid.Column widthMD={Columns.Four}>
+                        <Form.Label label="Idle Write Timeout (seconds)" />
+                        <LimitsField
+                          type={InputType.Number}
+                          name="timeout.queryidleWriteTimeoutSeconds"
+                          limits={limits}
+                          onChangeLimits={setLimits}
+                        />
+                      </Grid.Column>
+                    </Grid.Row>
+                  )}
+                  {limits?.stack && (
+                    <Grid.Row>
+                      <Grid.Column widthMD={Columns.Four}>
+                        <Form.Label label="Stack" />
+                        <LimitsField
+                          type={InputType.Text}
+                          name="stack.enabled"
+                          limits={limits}
+                          onChangeLimits={setLimits}
```

Comment:
I'm having a little trouble understanding how `setLimits` is working here. It looks like the trace is:

1. In `LimitsField`, a change in the value triggers the `onChange` function, which
2. sets the value of `newLimits` to `set(name, newValue, limits)`. That doesn't make sense to me - I would think the syntax here should be `set(limits, name, newValue)` https://lodash.com/docs/4.17.15#set
3. Run `onChangeLimits(newLimits)`, which is the `setLimits` function in the operator context.
4. That appears to refresh the current limits with `handleGetLimits`.
5. When the user hits `Submit Changes` in `OrgOverlay.tsx`, it triggers the `updateLimits` function, which does the following:
6. `const backendLImits = fromDisplayLimits(limits)` and then
7. `handleUpdateLimits(orgID, backendLimits)`
8. Which runs `putOperatorOrgsLimits({orgId: id, data: updatedLimits})

TLDR, I'm a little confused why `set` is even working here if it's in the wrong order, so I want to make sure we're not missing something on how the org limit update occurs.

Reply:
Your trace is correct. And the syntax for lodash's set is `object, path, value`. But if you take a look at the LodashSet interface declared in the node modules, the syntax is `<T extends object>(path: lodash.PropertyPath, value: any, object: T): T;` - which is how the LimitsField component is using it. We're passing the path we want to modify first, then the value we want to change it to, followed by the object itself.  This is precisely how we want to update timeout and stack limits as well.

Reply:
👍